sdk/storage/Azure.Storage.Blobs.Batching/src/BlobBatch.cs

```diff
@@ -0,0 +1,294 @@
+        internal bool IsAssociatedClient(BlobBatchClient client) =>
+            _client == client;
+
+        /// <summary>
+        /// Verify that the desired operation is allowed.
+        /// </summary>
+        /// <param name="operationType">
+        /// The type of operation to perform.
+        /// </param>
+        private void EnsureAllowed(BlobBatchOperationType operationType)
+        {
+            if (Submitted)
+            {
+                throw BatchErrors.BatchAlreadySubmitted();
+            }
+            else if (_operationType != null && _operationType != operationType)
+            {
+                throw BatchErrors.OnlyHomogenousOperationsAllowed(_operationType.Value);
+            }
+            _operationType = operationType;
+        }
+
+        #region DeleteBlob
+        /// <summary>
+        /// The <see cref="DeleteBlob(string, string, DeleteSnapshotsOption?, BlobAccessConditions?)"/>
+        /// operation marks the specified blob or snapshot for  deletion. The
+        /// blob is later deleted during garbage collection.
+        ///
+        /// Note that in order to delete a blob, you must delete all of its
+        /// snapshots. You can delete both at the same time using
+        /// <see cref="DeleteSnapshotsOption.Include"/>.
+        ///
+        /// For more information, see
+        /// <see href="https://docs.microsoft.com/rest/api/storageservices/delete-blob" />.
+        /// </summary>
+        /// <param name="containerName">
+        /// The name of the container containing the blob to delete.
+        /// </param>
+        /// <param name="blobName">
+        /// The name of the blob to delete.
+        /// </param>
+        /// <param name="deleteOptions">
+        /// Specifies options for deleting blob snapshots.
+        /// </param>
+        /// <param name="accessConditions">
+        /// Optional <see cref="BlobAccessConditions"/> to add conditions on
+        /// deleting this blob.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response"/> on successfully deleting.  The response
+        /// cannot be used until the batch has been submitted with
+        /// <see cref="BlobBatchClient.SubmitBatchAsync"/>.
+        /// </returns>
+        public virtual Response DeleteBlob(
+            string containerName,
+            string blobName,
+            DeleteSnapshotsOption? deleteOptions = default,
+            BlobAccessConditions? accessConditions = default)
+        {
+            var blobUri = new BlobUriBuilder(_client.Uri)
+            {
+                BlobContainerName = containerName,
+                BlobName = blobName
+            };
+            return DeleteBlob(
+                blobUri.ToUri(),
+                deleteOptions,
+                accessConditions);
+        }
+
+        /// <summary>
+        /// The <see cref="DeleteBlob(Uri, DeleteSnapshotsOption?, BlobAccessConditions?)"/>
```

Comment:
This looks like a copy from BlobBaseClient.Delete. Should we add information about how this won't actually do anything until the batch is submitted?

Reply:
Not fixing now - all docs need a lot of love next Preview